index.js

```diff
@@ -0,0 +1,163 @@
+'use strict';
+const mongo = require('mongodb');
+const async = require('async');
+const _ = require('lodash');
+const automap = require('automap');
+const defaults = {
+  mongo: {
+    host: 'mongodb://localhost:27017',
+    collectionName: 'hapi-activities'
+  },
+  timeout: 30 * 1000, // max time an action can take, default is 30 secs, set to false for infinity
+  interval: 5 * 60 * 1000, // 5 minutes
+  log: false
+};
+
+exports.register = (server, options, next) => {
+  const settings = _.defaults(options, defaults);
+  // connect to db:
+  mongo.connect(settings.mongo.host, (err, db) => {
+    if (err) {
+      return next(err);
+    }
+    // initialize the server object:
+    const collection = db.collection(settings.mongo.collectionName);
+
+    // update all activities:
+    const updateActivities = (allDone) => {
+      automap(
+        // fetch all 'waiting' activities
+        (done) => {
+          collection
+          .find({ status: 'waiting' })
+          .toArray((dbErr, results) => {
+            if (dbErr) {
+              server.log(['hapi-activities', 'error'], dbErr);
+            }
+            // can go back to sleep if nothing was found:
+            if (results.length === 0) {
+              return allDone();
+            }
+            done(null, results);
+          });
+        },
+        // for each activity we just fetched, do the following process:
+        (activity) => {
+          return {
+            // log that it's underway:
+            logActivity: (done) => {
+              if (settings.log) {
+                server.log(['hapi-activities', 'starting-activity', 'debug'], { message: 'Processing underway for activity', data: activity });
+              }
+              collection.update({ _id: activity._id }, { $set: { status: 'processing' } }, () => {
+                done();
+              });
+            },
+            // execute the actions in parallel:
+            performActions: ['logActivity', (results, done) => {
+              // will launch the activity's actions in parallel:
+              const updatedActivity = {
+                results: []
+              };
+              async.each(settings.activities[activity.activityName], (action, eachDone) => {
+                let actionData = activity.activityData;
+                // merge any default parameters for this action:
+                if (typeof action === 'object') {
+                  actionData = _.defaults(actionData, action.data);
+                  action = action.method;
+                }
+                // if a timeout is specified then put a timeout wrapper around the server method call:
+                const actionCall = settings.timeout ? async.timeout(server.methods[action], settings.timeout) : server.methods[action];
+                // now make the call:
+                try {
```

Comment:
If I take the try...catch expression out it will fail the test at https://github.com/firstandthird/hapi-activities/pull/1/files#diff-902f15dcedeada8279cb48f36e47bca0R144.  Does hapi-activities need to handle those sorts of errors?